mule-extensions-api/src/main/java/org/mule/runtime/extension/api/runtime/source/OnTerminateResult.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.runtime.source;
+
+import org.mule.runtime.extension.api.annotation.execution.OnError;
+import org.mule.runtime.extension.api.annotation.execution.OnSuccess;
+import org.mule.runtime.extension.api.annotation.execution.OnTerminate;
+import org.mule.runtime.api.message.Error;
+
+import java.util.Optional;
+
+/**
+ * Result for the {@link OnTerminate} callbacks. Provides information the result of the Source
```

Comment:
`Provides information the result of the Source generated message processing execution.` -> Besides grammar being funky, first describe what the class represents. Then say that it is used on the onTerminate callbacks.